robottelo/ui/computeresource.py

```diff
@@ -441,3 +442,25 @@ def set_profile_values(self, resource_name, profile_name, **kwargs):
             self, resource_type)
         resource_profile_form.set_values(**kwargs)
         resource_profile_form.submit()
+
+
+def host_provisioning_check(self, ip_addr):
+    """Check the provisioned host status by pinging the ip of host and check
+    to connect to ssh port
+
+    :param ip_addr: IP address of the provisioned host
+    :return: ssh command return code and stdout
+    """
+    result = ssh.command(
+        u'for i in {{1..60}}; do ping -c1 {0} && exit 0; sleep 20;'
+        u' done; exit 1'.format(ip_addr))
+    if result.return_code != 0:
+        raise (
+            'Failed to ping virtual machine Error:{0}'.format(result.stdout))
+    ssh_check = ssh.command(
+        u'for i in {{1..60}}; do nc -vn {0} 22 <<< "" && exit 0; sleep 20;'
+        u' done; exit 1'.format(ip_addr))
+    if ssh_check.return_code != 0:
+        raise (
+            'Failed to connect to SSH port of the virtual machine')
+    return ssh_check.return_code, ssh_check.stdout
```

Comment:
if this function fail, it will raise an error and the stdout is not useful simply think simply return True will be enough
also it will be nice to create and exception error like 
```python
class ProvisioningCheckError(Exception):
    pass
```
and raise the above error, this way in the tests you can use self.assertNotRaises(ProvisioningCheckError)
to be sure the exception is not raised